packages/typescript-estree/src/create-program/createProjectService.ts

```diff
@@ -0,0 +1,70 @@
+import * as ts from 'typescript';
+import * as tsserver from 'typescript/lib/tsserverlibrary';
```

Comment:
Can tell you at a glance that this worries me; you you effectively always need to use exactly one of these imports.

Reply:
Heh, it'd only been used in this file for `...ts.sys` and some types. Switched to always `typescript/lib/tsserverlibrary`. But in other files we're still `from 'typescript'`. Even if were were to completely switch over all our files, community plugins would still want to be able to import from `typescript`. 🤔

Reply:
It's tough, and I'm not sure what the answer is without a big shift.

If you were doing a plugin model like TS, you'd provide the TS API via a property on `context`, that way the version of TS that is being used by any given rule is exactly the one that is being used by eslint, which _sounds_ correct already, and would let you pass in the `tsseverlibrary` object too.

Reply:
Interestingly, we're already moving towards wrapping TypeScript APIs with our own services (e.g. https://github.com/typescript-eslint/typescript-eslint/pull/6404). So that might coincidentally make many rules out-of-the-box. Though I worry about edge cases around folks importing from `typescript` themselves...